### PR TITLE
feat(cdp): Added kinesis template

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by mariusandra** [comment]
This is ready to review now... From me? 😬 

I made a bunch of changes, some not too-related:

- Improved/simplified the `for-in` loop's bytecode, got rid of the magic "6 or 7" clause.
- Added a border to the "Done" button when editing a hog template input
- Added a bunch of array functions like `arrayPushBack`, `arraySort`, `arrayStringConcat`. They are all functional -> they don't modify the input array, and just return a newly constructed array with the changes. Why like this instead of mutating the array? In short, ClickHouse and unexpected effects in HogQL filters. At least the functions work the same.
- Added some string functions like `trim` and `splitByString`
- Added some crypto functions: `md5Hex`, `sha256Hex`, `sha256HmacChainHex` <-- these are not the same names as CH uses. Effectively ClickHouse's `lower(hex(MD5(s)))` equals `md5Hex(s)`. I didn't feel like introducing binary/bytes/buffer datatypes just for this.
- Fixed a bug in the Python VM with initializing empty `[]` arrays inside functions.
- Fixed a bug with `now()` in the Node VM

oh and also:

- Made the AWS Kinesis template work
- ... and added a test for it
- Plus added a way to mock `now()` in Hog template tests.

**Review by mariusandra — APPROVED** [review]
I approve of my changes

File: posthog/cdp/templates/helpers.py [hunk 1]
```diff
@@ -17,6 +17,11 @@ def setUp(self):
         super().setUp()
         self.compiled_hog = compile_hog(self.template.hog, supported_functions={"fetch", "print", "postHogCapture"})
 
+        # inject a stable now() function
+        frozen_time = "2024-04-16T12:34:51Z"
+        freeze_time = compile_hog("fn now() { return toDateTime('" + frozen_time + "') }")
+        self.compiled_hog = freeze_time + self.compiled_hog[1:]
```
Comment (benjackwhite) [inline posthog/cdp/templates/helpers.py:23]: cool

Comment (benjackwhite) [reply posthog/cdp/templates/helpers.py:23]: _although_ freezetime could also do this for us right?

**Review by benjackwhite — COMMENTED** [review]
Love it

**Pull request merged**